### DISCUSSION TITLE
Question: Which latest branch in imgui and node editor to avoid build issue( BeginHorizontal/BeginVertical/EndHorizontal/EndVertical/Spring)

### DISCUSSION THREAD

**Discussion by AkashSinha007** [ocornut/imgui#7384] [New Users Build/Link/Run issues ONLY!]
First off all thank you very much to @ocornut & @thedmd , I am able to think about node editor in C++.
I am fairly new to ImGui but finding it really interesting.

I am trying to compile the blueprint example on @thedmd 's repo for node-editor ( https://github.com/thedmd/imgui-node-editor/ ).
But I am getting compilation error due to missing definitions of BeginHorizontal/BeginVertical/EndHorizontal/EndVertical/Spring.

I tried to follow previous threads : 
https://github.com/ocornut/imgui/pull/846
https://github.com/thedmd/imgui-node-editor/issues/94

But I am not sure which branch in node-editor-repo and which branch in imgui repo I should consider for sucessful compilation.
Based on discussion on thread , I felt I should consider "develop" branch compared to "master" of node-editor repo. But I failed with same as well. 

I am using Meson Build for ImGui and was able to compile the "simple-example" under imgui-node-editor example sucessfully on Mac M1 only( I am yet to translate CMake to Meson for other platforms).
Refer : https://github.com/AkashSinha007/ImGui_NodeEditor_MesonBuild

The WrapDB in Meson Build helps in fetching imgui's source code and compiling automatically to generate dynamic libraries and all the interfaces to include directories/linking. 
It is using 1.89.9-1 version of ImGui . Please refer : https://github.com/AkashSinha007/ImGui_NodeEditor_MesonBuild/blob/master/subprojects/imgui.wrap

I want to make similar folder structure and try to make the compilation success with imgui-1.89.9-1 so that it is at latest.
Can you please guide on this compilation error ? Which version and branch I should be considering for each repo?

![Dependency_Error_ImGui_NodeEditor](https://github.com/ocornut/imgui/assets/75009548/efa8d50d-b003-4e59-ad30-d560a12f965a)

**Comment by AkashSinha007** [comment]
Please close this query . After properly reading through the source files I realized  the changes are made on imgui.cpp and associated files. I was getting confused as I was using higher versions of ImGui Library that I thought had decoupled code from node editor library. But Now I am clear. Thanks again !

**Reply by thedmd** [reply]
Please take look at node editor readme. BeginHorizontal is a [PR 846](https://github.com/ocornut/imgui/pull/846) to ImGui. It was born out of need to easily layout pins in blueprint nodes example. This PR is only required by an example, node editor will work with vanilla copy of the ImGui.
